recipes/armadillo/all/conanfile.py

```diff
@@ -0,0 +1,414 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class ArmadilloConan(ConanFile):
+    name = "armadillo"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://arma.sourceforge.net"
+    description = "Armadillo is a high quality C++ library for linear algebra and scientific computing, aiming towards a good balance between speed and ease of use."
+    topics = (
+        "linear algebra",
+        "scientific computing",
+        "matrix",
+        "vector",
+        "math",
+        "blas",
+        "lapack",
+        "mkl",
+        "hdf5",
+    )
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "use_blas": [
+            False,
+            "openblas",
+            "system_openblas",
+            "system_blas",
+            "system_mkl",
+            "system_flexiblas",
+            "framework_accelerate",
+        ],
+        "use_lapack": [
+            False,
+            "openblas",
+            "system_openblas",
+            "system_lapack",
+            "system_mkl",
+            "system_atlas",
+            "framework_accelerate",
+        ],
+        "use_hdf5": [False, "hdf5", "system_hdf5"],
+        "use_superlu": [False, "system_superlu"],
+        "use_extern_rng": [True, False],
+        "use_arpack": [False, "system_arpack"],
+        "use_wrapper": [True, False],
+        "mkl_library_path": "ANY",
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "use_blas": "openblas",
+        "use_lapack": False,
+        "use_hdf5": "hdf5",
+        "use_superlu": False,
+        "use_extern_rng": False,
+        "use_arpack": False,
+        "use_wrapper": False,
+        "mkl_library_path": "",
+    }
+    # Values that must be set for multiple options to be valid
+    co_dependencies = {
```

Comment:
```suggestion
    _co_dependencies = {
```

Any custom attribute, property, method, etc must be protected to avoid any possible name collision with future Conan versions.